x/merkledb/db.go

```diff
@@ -742,29 +741,53 @@ func (db *merkleDB) NewIteratorWithStartAndPrefix(start, prefix []byte) database
 // the movement of [node] from [db.nodeCache] to [db.nodeDB] is atomic.
 // As soon as [db.nodeCache] no longer has [node], [db.nodeDB] does.
 // Non-nil error is fatal -- causes [db] to close.
-func (db *merkleDB) onEviction(node *node) error {
-	if node == nil || node.hasValue() {
```

Comment:
why did we stop checking `node.hasValue()`? did we mean to include this check in `writeNodeToBatch` as the comment suggests?